spring-batch-integration/src/main/java/org/springframework/batch/integration/chunk/RemoteChunkingMasterStepBuilder.java

```diff
@@ -0,0 +1,426 @@
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.interceptor.TransactionAttribute;
+import org.springframework.util.Assert;
+
+import java.util.function.Function;
+
+/**
+ * Builder for a master step in a remote chunking setup. This builder:
+ *
+ * <ul>
+ *     <li>creates and sets a {@link ChunkMessageChannelItemWriter} on the master step</li>
+ *     <li>registers a {@link RemoteChunkHandlerFactoryBean} in the given context</li>
+ * </ul>
+ *
+ * @param <I> type of input items
+ * @param <O> type of output items
+ *
+ * @since 4.1
+ * @author Mahmoud Ben Hassine
+ */
+public class RemoteChunkingMasterStepBuilder<I, O> extends FaultTolerantStepBuilder<I, O> {
+
+	private MessagingTemplate messagingTemplate;
+	private GenericApplicationContext applicationContext;
+	private PollableChannel inputChannel;
+	private DirectChannel outputChannel;
+
+	private final int DEFAULT_MAX_WAIT_TIMEOUTS = 40;
+	private static final long DEFAULT_THROTTLE_LIMIT = 6;
+	private int maxWaitTimeouts = DEFAULT_MAX_WAIT_TIMEOUTS;
+	private long throttleLimit = DEFAULT_THROTTLE_LIMIT;
+
+	/**
+	 * Create a new {@link RemoteChunkingMasterStepBuilder}.
+	 *
+	 * @param stepName name of the master step
+	 */
+	public RemoteChunkingMasterStepBuilder(String stepName) {
+		super(new StepBuilder(stepName));
+	}
+
+	/**
+	 * Set the application context in which required beans will be registered.
+	 *
+	 * @param applicationContext the application context to set.
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemoteChunkingMasterStepBuilder<I, O> applicationContext(GenericApplicationContext applicationContext) {
+		Assert.notNull(applicationContext, "applicationContext must not be null");
+		this.applicationContext = applicationContext;
+		return this;
+	}
+
+	/**
+	 * Set the input channel on which replies from workers will be received.
+	 *
+	 * @param inputChannel the input channel
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemoteChunkingMasterStepBuilder<I, O> inputChannel(PollableChannel inputChannel) {
+		Assert.notNull(inputChannel, "inputChannel must not be null");
+		this.inputChannel = inputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the output channel on which requests to workers will be sent.
+	 *
+	 * @param outputChannel the output channel.
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemoteChunkingMasterStepBuilder<I, O> outputChannel(DirectChannel outputChannel) {
+		Assert.notNull(outputChannel, "outputChannel must not be null");
+		this.outputChannel = outputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the {@link MessagingTemplate} to use to send data to workers.
+	 *
+	 * @param messagingTemplate the messaging template to use
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemoteChunkingMasterStepBuilder<I, O> messagingTemplate(MessagingTemplate messagingTemplate) {
```

Comment:
We need to be clear here that we are going to override the default destination for this instance in case a user accidentally shares it.

Reply:
I added this clarification.

Reply:
We probably should note at the class level (?) that if a `MessagingTemplate` is not provided, we will create one that uses the default values.  I can add this on merge if you agree.

Reply:
I agree. I added a clarification at the class level about this detail. See 0a7eda3a776917d9c89860d220bfc546f2e3370a